LeaderboardBackend.Test/Categories.cs

```diff
@@ -288,4 +289,649 @@ public static async Task CreateCategory_BadData(string? name, string? slug, obje
+                Body = new CreateCategoryRequest()
+                {
+                    Name = "Update Should Not Conflict Deleted",
+                    Slug = "updatecat-no-conflict-deleted",
+                    SortDirection = SortDirection.Ascending,
+                    Type = RunType.Score,
+                },
+                Jwt = _jwt,
+            }
+        );
+
+        await FluentActions.Awaiting(() => _apiClient.Patch(
+            $"category/{toNotConflict.Id}",
+            new()
+            {
+                Body = new UpdateCategoryRequest()
+                {
+                    Slug = "updatecat-deleted"
+                },
+                Jwt = _jwt
+            }
+        )).Should().NotThrowAsync();
+    }
+
+    [Test]
+    public static async Task UpdateCategory_NoConflictBecauseDifferentLeaderboard()
+    {
+        CategoryViewModel first = await _apiClient.Post<CategoryViewModel>(
+            $"leaderboard/{_createdLeaderboard.Id}/categories/create",
+            new()
+            {
+                Body = new CreateCategoryRequest()
+                {
+                    Name = "Update No Conflict",
+                    Slug = "updatecat-no-conflict-different-board",
+                    SortDirection = SortDirection.Ascending,
+                    Type = RunType.Score,
+                },
+                Jwt = _jwt,
+            }
+        );
+
+        LeaderboardViewModel board = await _apiClient.Post<LeaderboardViewModel>(
+            $"leaderboards/create",
+            new()
+            {
+                Body = new CreateLeaderboardRequest()
+                {
+                    Name = "Update Cat Different Board",
+                    Slug = "updatecat-no-conflict-different-board",
+                },
+                Jwt = _jwt,
+            }
+        );
+
+        board.Id.Should().NotBe(default);
+
+        CategoryViewModel toNotConflict = await _apiClient.Post<CategoryViewModel>(
+            $"leaderboard/{board.Id}/categories/create",
+            new()
+            {
+                Body = new CreateCategoryRequest()
+                {
+                    Name = "Should Not Conflict",
+                    Slug = "updatecat-should-not-conflict-different-board",
+                    SortDirection = SortDirection.Ascending,
+                    Type = RunType.Score,
+                },
+                Jwt = _jwt,
+            }
+        );
+
+        await FluentActions.Awaiting(() => _apiClient.Patch(
+            $"category/{toNotConflict.Id}",
+            new()
+            {
+                Body = new UpdateCategoryRequest()
+                {
+                    Slug = "updatecat-no-conflict-different-board",
+                },
+                Jwt = _jwt,
+            }
+        )).Should().NotThrowAsync();
+    }
+
+    [TestCase(1, "b.b")]
+    [TestCase(2, "b")]
+    [TestCase(3, null)]
+    public static async Task UpdateCategory_BadData(int index, string? slug)
```

Comment:
What is the `index` param doing here?

Reply:
To act as a differentiator for the cats I create. Can't add `slug` to the cats' slugs